lib/services/database.dart

```diff
@@ -55,4 +58,36 @@ class DatabaseService {
   Stream<DocumentSnapshot> get userData {
     return userDetails.document(uid).snapshots();
   }
+
+  // add group details
+  Future<void> createTrip(RequestDetails requestDetails) async {
+    var user = await _auth.currentUser();
+    final docRef = await requestdetails.add({
+      'destination': requestDetails.destination.toString(),
+      'startDate': requestDetails.startDate.toString(),
+      'startTime': requestDetails.startTime.toString(),
+      'endDate': requestDetails.endDate.toString(),
+      'endTime': requestDetails.endTime.toString(),
+      'privacy': requestDetails.privacy.toString(),
+    });
+
+    var request = requestdetails.document(docRef.documentID).collection('users');
+    await Firestore.instance.collection('userdetails').document(user.uid).get().then((value) async {
+      if (value.exists) {
+        await request.document(user.uid).setData({
+          'name': value.data['name'],
+          'hostel': value.data['hostel'],
+          'sex': value.data['sex'],
+          'mobilenum': value.data['mobileNumber'],
+          'totalrides': value.data['totalRides'],
+          'actualrating': value.data['actualRating'],
+          'cancelledrides': value.data['cancelledRides'],
+        });
+      }
+    });
```

Comment:
Just add userId. No need to add the entire user detail again to the group request

Reply:
Am commenting this set of code for now but we can think about this later as this will reduce the number of document reads as we wont have to fetch details again and again.

Reply:
i have added this back. I need it to display user info in the group page. pls review.